### PR TITLE
lantiq: xway: add support for AVM FRITZ!Box 7330

### PR DESCRIPTION
**Pull request by abajk**
The AVM FRITZ!Box 7330 shares hardware with the AVM Fritzbox 7320 except for the second ethernet port, which only supports 100M.

Hardware:
- SoC: Lantiq ARX 188
- CPU: 2x MIPS 34Kc 393 MHz
- RAM: 64 MiB 196 MHz
- Flash: 16 MiB NAND
- Ethernet: Built-in Gigabit Ethernet switch, 1x 1GbE, 1x 100M
- Wifi: Atheros AR9227-BC2A b/g/n with 2 pcb/internal antennas
- USB: 2x USB 2.0
- DSL: Built-in ADSL2+ modem
- DECT: Dialog SC14441
- LEDs: 1 two-color, 4 one-color
- Buttons: 1x DECT, 1x WIFI
- Telephone connectors: 1 FXS port via TAE or RJ11 connector

Installation:
The installation process is described on the wiki.

Unsupported (same as AVM 7320):
- VoIP (DECT and FXS),
- Second Ethernet port.

Signed-off-by: Aleksander Jan Bajkowski <olek2@wp.pl>

### CODE REVIEW

File: target/linux/lantiq/image/ar9.mk [hunk 1]
```diff
@@ -29,6 +29,21 @@ define Device/avm_fritz7320
 endef
 TARGET_DEVICES += avm_fritz7320
 
+define Device/avm_fritz7330
+  $(Device/AVM_preloader)
+  DEVICE_MODEL := FRITZ!Box 7330
+  SOC := ar9
+  IMAGE_SIZE := 15744k
+  LOADER_FLASH_OFFS := 0x31000
+  DEVICE_PACKAGES := kmod-ath9k kmod-owl-loader wpad-basic-mbedtls \
```
Comment (hauke) [inline target/linux/lantiq/image/ar9.mk:38]: The DTS file and these settings are the same except the name of the board. I would prefer to just add an  alt entry to the fritzbox 7320 entry. 
````
DEVICE_ALT1_VENDOR :=AVM 
DEVICE_ALT1_MODEL := FRITZ!Box 7330
````

File: target/linux/lantiq/image/ar9.mk [hunk 2]
```diff
@@ -29,6 +29,21 @@ define Device/avm_fritz7320
...
+	kmod-ltq-adsl-ar9-mei kmod-ltq-adsl-ar9 \
+	kmod-ltq-adsl-ar9-fw-b kmod-ltq-atm-ar9 \
+	ltq-adsl-app ppp-mod-pppoa \
+	kmod-ltq-deu-ar9 kmod-usb-dwc2 fritz-tffs -swconfig
+  SUPPORTED_DEVICES += FRITZ7330
```
Comment (hauke) [inline target/linux/lantiq/image/ar9.mk:43]: Why do you add a `SUPPORTED_DEVICES` entry? 
This is only needed when sysupgrade should also work for images named `SUPPORTED_DEVICES`.

File: target/linux/lantiq/image/ar9.mk (hunk 1 above)
Comment (abajk) [reply target/linux/lantiq/image/ar9.mk]: You are right that the dts for both devices are same. I expect they will differ in the future. I have currently followed the suggested approach and only added an alternative name for the AVM 7330. The dts files for both devices can be spliced in the future when necessary

File: target/linux/lantiq/image/ar9.mk (hunk 2 above)
Comment (abajk) [reply target/linux/lantiq/image/ar9.mk]: This comment is no longer relevant after applying the suggested changes, so I mark it as resolved.

**Pull request merged**